tests/wlr_layer_shell_v1.cpp

```diff
@@ -944,6 +944,65 @@ TEST_P(LayerSurfaceLayoutTest, maximized_xdg_toplevel_is_shrunk_for_exclusive_zo
     EXPECT_THAT(new_height, Eq(expected_height));
 }
 
+TEST_P(LayerSurfaceLayoutTest, surfaces_with_exclusive_zone_set_to_negative_one_do_not_respect_other_exclusive_zones)
+{
+    auto const layout = GetParam();
+    auto const request_size = layout.request_size();
+    auto const initial_rect = layout.placement_rect(output_rect());
+
+    zwlr_layer_surface_v1_set_anchor(layer_surface, layout);
+    zwlr_layer_surface_v1_set_exclusive_zone(layer_surface, -1);
+    invoke_zwlr_layer_surface_v1_set_margin(layer_surface, layout.margin);
+    zwlr_layer_surface_v1_set_size(layer_surface, request_size.width.as_int(), request_size.height.as_int());
+    commit_and_wait_for_configure();
+    surface.attach_visible_buffer(initial_rect.size.width.as_int(), initial_rect.size.height.as_int());
+
+    // First, let's check that the surface has the size and position that we would expect
+    client.roundtrip();
+
+    if (request_size.width.as_int())
+        EXPECT_THAT(layer_surface.last_size().width, Eq(request_size.width));
+    if (request_size.height.as_int())
+        EXPECT_THAT(layer_surface.last_size().height, Eq(request_size.height));
```

Comment:
I'm pretty sure this isn't testing what you expect it to be testing? Particularly: we've attached a buffer of size `initial_rect.size`. That means that the surface has size `initial_rect.size`; the client buffer size is authoritative.

What this is testing is that the WM has *suggested* (in the `configure` event) that we make the surface size `request_size.size`.